xds/internal/balancer/edsbalancer/xds_client_wrapper.go

```diff
@@ -48,40 +48,45 @@ var (
 )
 
 type loadStoreWrapper struct {
-	mu      sync.RWMutex
-	store   *load.Store
-	service string
+	mu         sync.RWMutex
+	store      *load.Store
+	service    string
+	perCluster load.PerClusterReporter
 }
 
 func (lsw *loadStoreWrapper) update(store *load.Store, service string) {
 	lsw.mu.Lock()
 	defer lsw.mu.Unlock()
+	if store == lsw.store && service == lsw.service {
```

Comment:
Not convinced if we need this check here and in the lrs/balancer.go.

Reply:
Do you mean we don't need this check? Or we should move this check somewhere else?

The purpose of this check is to use the cached `perCluster` to avoid getting the lock inside `store`. This update can happen frequently, with the same parameters.

Reply:
I thought we don't need this check because I thought this update will not happen frequently. Why would this update happen frequently with the same parameters?

Reply:
This is called on each received CDS response. (Not that frequently comparing to RPCs.)

A normal unnecessary (from lrs's point of view) update would be, something else (the clusters) in CDS changed, but the lrs-server string doesn't.
A not-so-normal unnecessary update is when the server sends the same CDS resp multiple times.